lte/gateway/c/core/oai/lib/n11/M5GMobilityServiceClient.cpp

```diff
@@ -71,14 +72,106 @@ static void handle_allocate_ipv4_address_status(
+  MessageDef* message_p;
+  message_p =
+      itti_alloc_new_message(TASK_GRPC_SERVICE, AMF_IP_ALLOCATION_RESPONSE);
+
+  itti_amf_ip_allocation_response_t* amf_ip_allocation_response_p;
+  amf_ip_allocation_response_p = &message_p->ittiMsg.amf_ip_allocation_response;
+  memset(
+      amf_ip_allocation_response_p, 0,
+      sizeof(itti_amf_ip_allocation_response_t));
+
+  memcpy(amf_ip_allocation_response_p->imsi, imsi, IMSI_BCD_DIGITS_MAX);
+  amf_ip_allocation_response_p->imsi_length      = IMSI_BCD_DIGITS_MAX;
+  amf_ip_allocation_response_p->pdu_session_id   = pdu_session_id;
+  amf_ip_allocation_response_p->pti              = pti;
+  amf_ip_allocation_response_p->pdu_session_type = pdu_session_type;
+  amf_ip_allocation_response_p->paa.ipv6_address = in_ip6_addr;
+  amf_ip_allocation_response_p->paa.pdn_type     = IPv6;
+  amf_ip_allocation_response_p->paa.vlan         = vlan;
+
+  memcpy(
+      amf_ip_allocation_response_p->gnb_gtp_teid_ip_addr, gnb_gtp_teid_ip_addr,
+      4);
+
+  memcpy(amf_ip_allocation_response_p->apn, apn, strlen(apn) + 1);
+
+  if (status.ok()) {
+    amf_ip_allocation_response_p->result = SGI_STATUS_OK;
+  } else {
+    if (status.error_code() == grpc::StatusCode::ALREADY_EXISTS) {
+      amf_ip_allocation_response_p->result = SGI_STATUS_ERROR_SYSTEM_FAILURE;
+    } else {
+      amf_ip_allocation_response_p->result =
+          SGI_STATUS_ERROR_ALL_DYNAMIC_ADDRESSES_OCCUPIED;
+    }
+  }
+
+  send_msg_to_task(&grpc_service_task_zmq_ctx, TASK_AMF_APP, message_p);
+}
+
+static void handle_allocate_ipv4v6_address_status(
+    const grpc::Status& status, struct in_addr in_ip4_addr,
+    struct in6_addr in_ip6_addr, int vlan, const char* imsi, const char* apn,
+    uint32_t pdu_session_id, uint8_t pti, uint32_t pdu_session_type,
+    uint32_t gnb_gtp_teid, uint8_t* gnb_gtp_teid_ip_addr,
+    uint8_t gnb_gtp_teid_ip_addr_len) {
+  MessageDef* message_p;
+  message_p =
+      itti_alloc_new_message(TASK_GRPC_SERVICE, AMF_IP_ALLOCATION_RESPONSE);
+
+  itti_amf_ip_allocation_response_t* amf_ip_allocation_response_p;
+  amf_ip_allocation_response_p = &message_p->ittiMsg.amf_ip_allocation_response;
+  memset(
+      amf_ip_allocation_response_p, 0,
+      sizeof(itti_amf_ip_allocation_response_t));
+  memcpy(amf_ip_allocation_response_p->imsi, imsi, IMSI_BCD_DIGITS_MAX);
+  amf_ip_allocation_response_p->imsi_length      = IMSI_BCD_DIGITS_MAX;
+  amf_ip_allocation_response_p->pdu_session_id   = pdu_session_id;
+  amf_ip_allocation_response_p->pti              = pti;
+  amf_ip_allocation_response_p->pdu_session_type = pdu_session_type;
+  amf_ip_allocation_response_p->paa.ipv4_address = in_ip4_addr;
+  amf_ip_allocation_response_p->paa.ipv6_address = in_ip6_addr;
+  amf_ip_allocation_response_p->paa.pdn_type     = IPv4_AND_v6;
+  amf_ip_allocation_response_p->paa.vlan         = vlan;
+  memcpy(
+      amf_ip_allocation_response_p->gnb_gtp_teid_ip_addr, gnb_gtp_teid_ip_addr,
+      4);
```

Comment:
You could use You could use gnb_gtp_teid_ip_addr_len.

Reply:
Addressed review comment